source/common/http/functional_stream_decoder_base.cc

```diff
@@ -63,8 +70,13 @@ FunctionRetrieverMetadataAccessor::getRouteMetadata() const {
 
 absl::optional<FunctionRetrieverMetadataAccessor::Result>
 FunctionRetrieverMetadataAccessor::tryToGetSpec() {
-  const Router::RouteEntry *routeEntry =
-      SoloFilterUtility::resolveRouteEntry(decoder_callbacks_);
+
+  route_info_ = decoder_callbacks_->route();
+  if (!route_info_) {
+    return {};
+  }
+
+  const Router::RouteEntry *routeEntry = route_info_->routeEntry();
```

Comment:
What is the difference between the previous code that used `SoloFilterUtility::resolveRouteEntry()` and the current version? Is it just that `route_info_` is saved into a member?

Reply:
yes